### PR TITLE
Make collection of Sessions to be UniqueList

### LINKED ISSUE #344: Bug: User Guide `div` element missing closing `</div>` tag
image

### PR DESCRIPTION
**Pull request by njxue**
Sessions should be kept in UniqueList to be consistent with how ModuleClasses and and Students are organised.
Also added integration tests for Adds command
DG not updated yet

Also resolves #344

### CODE REVIEW

File: src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java [hunk 1]
```diff
@@ -27,7 +25,7 @@ public class ModuleClass implements Identity<ModuleClass>, Comparable<ModuleClas
 
     private final String className;
 
-    private final List<Session> sessions;
+    private final UniqueList<Session> sessions;
```
Comment (RezwanArefin01) [inline src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java]: For consistency, should this be changed as follows?
```suggestion
    private final UniqueList<Session> sessions = new UniqueList<>();
```

File: src/main/java/seedu/taassist/model/uniquelist/UniqueList.java [hunk 2]
```diff
@@ -64,6 +64,16 @@ public void addAll(Collection<T> toAddList) {
         Collections.sort(internalList);
     }
 
+    /**
+     * Adds all elements in the {@code toAddList} to the current list.
+     * The elements must not already exist in the list.
+     */
+    public void addAll(UniqueList<T> toAddUniqueList) {
```
Comment (RezwanArefin01) [inline src/main/java/seedu/taassist/model/uniquelist/UniqueList.java]: Is there a need for this method? All current usages seem to be able to cover by `setElements`.

File: src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java [hunk 3]
```diff
@@ -48,12 +46,27 @@ public ModuleClass(String className) {
      * @param sessions A list of sessions.
      */
     public ModuleClass(String className, List<Session> sessions) {
+        requireAllNonNull(className, sessions);
+        checkArgument(isValidModuleClassName(className), MESSAGE_CONSTRAINTS);
+        this.className = className;
+        this.sessions = new UniqueList<>();
+        this.sessions.addAll(sessions);
```
Comment (RezwanArefin01) [inline src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java]: After the above change:
```suggestion
        this.sessions.setElements(sessions);
```

File: src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java [hunk 4]
```diff
@@ -38,7 +36,7 @@ public ModuleClass(String className) {
         requireNonNull(className);
         checkArgument(isValidModuleClassName(className), MESSAGE_CONSTRAINTS);
         this.className = className;
-        sessions = new ArrayList<Session>();
+        this.sessions = new UniqueList<>();
```
Comment (RezwanArefin01) [inline src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java]: After above change:
```suggestion
```

File: src/test/java/seedu/taassist/logic/commands/AddsCommandTest.java [hunk 5]
```diff
@@ -65,6 +77,66 @@ public void execute_validSessionsWithDuplicates_success() throws CommandExceptio
...
+    public void execute_addSingleNewSession_success() throws CommandException {
+        ModuleClass focusedClass = model.getFocusedClass();
+        Session validSession = new SessionBuilder().withName(VALID_SESSION_LAB1).build();
+        assert !focusedClass.hasSession(validSession);
+
+        Set<Session> validSessions = new HashSet<>(List.of(validSession));
+        AddsCommand command = new AddsCommand(validSessions);
+        command.execute(expectedModel);
+
+        assertCommandSuccess(command, model, AddsCommand.getCommandMessage(validSessions, new HashSet<>()),
```
Comment (RezwanArefin01) [inline src/test/java/seedu/taassist/logic/commands/AddsCommandTest.java:99]: Should `getCommandMessage` have public visibility? I'd prefer if a hardcoded message is constructed here and the method is set to have private visibility.

Comment (RezwanArefin01) [inline src/test/java/seedu/taassist/logic/commands/AddsCommandTest.java:99]: Should `getCommandMessage` have public visibility? I'd prefer if a hard-coded message is constructed here and the method and other similar methods are set to have private visibility. 

Also if these methods are public, should there be tests for them?

**Review by RezwanArefin01 — CHANGES_REQUESTED** [review]
Suggested some consistency changes.

File: src/test/java/seedu/taassist/logic/commands/AddsCommandTest.java (hunk 5 above)
Comment (njxue) [reply src/test/java/seedu/taassist/logic/commands/AddsCommandTest.java:99]: @RezwanArefin01 If we hard code the message, then it's going to be quite tedious to properly format the message in test cases and can potentially cause code duplication in many places. So I think the preferred solution is to keep them public and add tests for them (I will do so in another issue).

Also, some commands have `getCommandMessage` while some don't, which is inconsistent. Do you think we should make all commands have/not have the method? Or it's okay to leave it as it is?

Comment (RezwanArefin01) [reply src/test/java/seedu/taassist/logic/commands/AddsCommandTest.java:99]: @njxue I think it is fine the way it currently is.

Comment (njxue) [reply src/test/java/seedu/taassist/logic/commands/AddsCommandTest.java:99]: Ok, will create new issue regarding tests for `getCommandMessage`

File: src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java [hunk 6]
```diff
@@ -70,13 +82,12 @@ public String getClassName() {
      * if modification is attempted.
      */
     public List<Session> getSessions() {
```
Comment (RezwanArefin01) [inline src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java]: I think `List<Session>` should be `ObservableList<Session>`. Return type should not limit functionality. :thinking: 

Can you also check in other places where observable list is returned as list?

File: src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java [hunk 7]
```diff
@@ -51,13 +48,28 @@ public ModuleClass(String className, List<Session> sessions) {
...
+        this.sessions.setElements(sessions);
     }
 
+    /**
+     * Constructs a {@code ModuleClass} with the provided unique list of {@code Session}-s.
+     *
+     * @param className A valid class name.
+     * @param sessions A unique list of sessions.
+     */
+    public ModuleClass(String className, UniqueList<Session> sessions) {
```
Comment (RezwanArefin01) [inline src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java]: Is this constructor needed? I think current usages can be done with `List` only.

**Review by RezwanArefin01 — CHANGES_REQUESTED** [review]
Some consistency changes suggested.

File: src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java (hunk 6 above)
Comment (njxue) [reply src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java]: ok, other parts of the code should also be fixed

**Review by RezwanArefin01 — APPROVED** [review]
Looks good to merge!

**Pull request merged**